iceoryx2-bb/cxx/include/iox2/container/static_string.hpp

```diff
@@ -0,0 +1,425 @@
+// Copyright (c) 2025 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+#ifndef IOX2_INCLUDE_GUARD_CONTAINER_STATIC_STRING_HPP
+#define IOX2_INCLUDE_GUARD_CONTAINER_STATIC_STRING_HPP
+
+#include "iox2/container/config.hpp"
+#include "iox2/container/optional.hpp"
+
+#include <algorithm>
+#include <cstddef>
+#include <cstdint>
+#include <cstring>
+#include <functional>
+#include <memory>
+#include <type_traits>
+
+namespace iox2 {
+namespace container {
+
+/// A UTF-8 string with fixed static capacity and contiguous inplace storage.
+/// The string class uses Unicode (ISO/IEC 10646) terminology throughout its interface. In particular:
+/// - A code point is the numerical index assigned to a character in the Unicode standard.
+/// - A code unit is the basic component of a character encoding system. For UTF-8, the code unit has a size of 8-bits
+/// For example, the code point U+0041 represents the letter 'A' and can be encoded in a single 8-bit code unit in
+/// UTF-8. The code point U+1F4A9 requires four 8-bit code units in the UTF-8 encoding.
+///
+/// @attention The NUL code point (U+0000) is not allowed anywhere in the string.
+/// @note Currently only Unicode code points less than 128 (U+0080) are supported.
+///       This restricts the valid contents of a string to those UTF8 strings
+///       that are also valid 7-bit ASCII strings. Full Unicode support will get added later.
+/// @tparam N Maximum number of UTF-8 code units that the string can store, excluding the terminating NUL character.
+template <uint64_t N>
+class StaticString {
```

Comment:
Is there a reason not to add the string to the release notes yet?

Reply:
Nope, I wanted to do that after rebasing the branch on main, to avoid triggering a merge conflict. So if everything else looks good, I can go ahead with that.